### PR TITLE
Renames `provider_uri` to `eth_provider_uri`

### PR DESCRIPTION
**Pull request by vzotova**
**Type of PR:**
- Other

**Required reviews:** 
- 3

### CODE REVIEW

**Comment by derekpierre** [comment]
I assume the impetus for this PR is the conversation around us now having 3 providers:
```
--provider.   # ETH node provider
--payment-provider.   # L2 provider
--staking-provider  #  Threshold Staking
```

Just to confirm, we are good with the following end result after this PR:
```
--eth-provider   # ETH node provider
--payment-provider   # L2 provider
--staking-provider  # Threshold Staking
```

One tricky notion (although perhaps minor) is that the value for `--staking-provider` is an address whereas the other two are URIs.

**Comment by vzotova** [comment]
> I assume the impetus for this PR is the conversation around us now having 3 providers:
> 
> ```
> --provider.   # ETH node provider
> --payment-provider.   # L2 provider
> --staking-provider  #  Threshold Staking
> ```
> 
> Just to confirm, we are good with the following end result after this PR:
> 
> ```
> --eth-provider   # ETH node provider
> --payment-provider   # L2 provider
> --staking-provider  # Threshold Staking
> ```
> 
> One tricky notion (although perhaps minor (?)) is that the value for `--staking-provider` is an address whereas the other two provider values are URIs.

Also in this PR possible:
```
--eth-network-provider   # ETH node provider
--payment-network-provider   # L2 provider
--staking-provider  # Threshold Staking
```

And yes, staking is address, eth/payment is URI

**Comment by KPrasch** [comment]
Of course we can ass shorthand variants, but `--payment-network-provider` is too long IMO.

**Comment by KPrasch** [comment]
A few minor conflicts arose from merging #2862

**Review by vepkenez — APPROVED** [review]

**Review by KPrasch — APPROVED** [review]

File: nucypher/cli/commands/porter.py [hunk 1]
```diff
@@ -140,7 +140,7 @@ def run(general_config,
 
     emitter.message(f"Network: {PORTER.domain.capitalize()}", color='green')
     if not federated_only:
-        emitter.message(f"Provider: {provider_uri}", color='green')
+        emitter.message(f"Provider: {eth_provider_uri}", color='green')
```
Comment (derekpierre) [inline nucypher/cli/commands/porter.py:143]:
```suggestion
        emitter.message(f"ETH Provider URI: {eth_provider_uri}", color='green')
```

File: nucypher/cli/commands/stake.py [hunk 2]
```diff
@@ -190,10 +190,10 @@ def retrieve_config(self, emitter, config_file):
 
     def generate_config(self, config_root):
 
-        if self.provider_uri is None:
+        if self.eth_provider_uri is None:
             raise click.BadOptionUsage(
-                option_name="--provider",
-                message="--provider must be specified to create a new stakeholder")
+                option_name="--eth-network-provider",
+                message="--eth-network-provider must be specified to create a new stakeholder")
```
Comment (derekpierre) [inline nucypher/cli/commands/stake.py:196]:
```suggestion
                option_name="--eth-provider",
                message="--eth-provider must be specified to create a new stakeholder")
```

File: nucypher/cli/commands/ursula.py [hunk 3]
```diff
@@ -326,8 +326,8 @@ def init(general_config, config_options, force, config_root, key_material):
     _pre_launch_warnings(emitter, dev=None, force=force)
     if not config_root:
         config_root = general_config.config_root
-    if not config_options.federated_only and not config_options.provider_uri:
-        raise click.BadOptionUsage('--provider', message="--provider is required to initialize a new ursula.")
+    if not config_options.federated_only and not config_options.eth_provider_uri:
+        raise click.BadOptionUsage('--eth-network-provider', message="--eth-network-provider is required to initialize a new ursula.")
```
Comment (derekpierre) [inline nucypher/cli/commands/ursula.py:330]:
```suggestion
        raise click.BadOptionUsage('--eth-provider', message="--eth-provider is required to initialize a new ursula.")
```

**Review by derekpierre — COMMENTED** [review]
Thanks for taking this on @vzotova 🤗

File: nucypher/cli/options.py [hunk 4]
```diff
@@ -57,7 +57,7 @@
 option_min_stake = click.option('--min-stake', help="The minimum stake the teacher must have to be locally accepted.", type=STAKED_TOKENS_RANGE, default=MIN_AUTHORIZATION)
 option_parameters = click.option('--parameters', help="Filepath to a JSON file containing additional parameters", type=EXISTING_READABLE_FILE)
 option_participant_address = click.option('--participant-address', help="Participant's checksum address.", type=EIP55_CHECKSUM_ADDRESS)
-option_payment_provider = click.option('--payment-provider', help="Connection URL for payment method", type=click.STRING, required=False)
+option_payment_provider = click.option('--payment-network-provider', '--payment-provider', 'payment_provider', help="Connection URL for payment method", type=click.STRING, required=False)
```
[the remark this answers is not in the document]
Comment (vzotova) [reply nucypher/cli/options.py]: ✔️

File: nucypher/cli/painting/deployment.py [hunk 5]
```diff
@@ -38,7 +38,7 @@ def paint_staged_deployment(emitter, deployer_interface, administrator) -> None:
     emitter.clear()
     emitter.banner(NU_BANNER)
     emitter.echo(f"Current Time ........ {maya.now().iso8601()}")
-    emitter.echo(f"Web3 Provider ....... {deployer_interface.provider_uri}")
+    emitter.echo(f"Web3 Provider ....... {deployer_interface.eth_provider_uri}")
```
[the remark this answers is not in the document]
Comment (vzotova) [reply nucypher/cli/painting/deployment.py]: ✔️

File: nucypher/cli/painting/deployment.py [hunk 6]
```diff
@@ -101,7 +101,7 @@ def paint_deployer_contract_inspection(emitter, registry, deployer_address) -> N
 * Web3 Provider
 ====================================================================
 
-Provider URI ............. {blockchain.provider_uri}
+Provider URI ............. {blockchain.eth_provider_uri}
```
[the remark this answers is not in the document]
Comment (vzotova) [reply nucypher/cli/painting/deployment.py]: ✔️

File: nucypher/cli/commands/stake.py (hunk 2 above)
Comment (vzotova) [reply nucypher/cli/commands/stake.py]: ✔️

File: nucypher/cli/commands/ursula.py (hunk 3 above)
Comment (vzotova) [reply nucypher/cli/commands/ursula.py]: ✔️

File: tests/acceptance/cli/test_porter.py [hunk 7]
```diff
@@ -278,7 +278,7 @@ def test_blockchain_porter_cli_run_provider_required(click_runner,
     result = click_runner.invoke(nucypher_cli, porter_run_command, catch_exceptions=False)
 
     assert result.exit_code != 0
-    assert "--provider is required" in result.output
+    assert "--eth-network-provider is required" in result.output
```
[the remark this answers is not in the document]
Comment (vzotova) [reply tests/acceptance/cli/test_porter.py]: ✔️

File: tests/acceptance/cli/test_porter.py [hunk 8]
```diff
@@ -237,30 +237,30 @@ def test_blockchain_porter_cli_run_simple(click_runner,
...
-                          '--provider', TEST_PROVIDER_URI,
+                          '--eth-provider', TEST_ETH_PROVIDER_URI,
                           '--registry-filepath', agency_local_registry.filepath,
                           '--teacher', blockchain_teacher_uri)
     result = click_runner.invoke(nucypher_cli, porter_run_command, catch_exceptions=False)
     assert result.exit_code == 0
     output = result.output
     assert f"Network: {TEMPORARY_DOMAIN}" in output
-    assert f"Provider: {TEST_PROVIDER_URI}" in output
+    assert f"Provider: {TEST_ETH_PROVIDER_URI}" in output
```
[the remark this answers is not in the document]
Comment (vzotova) [reply tests/acceptance/cli/test_porter.py]: ✔️

**Review by derekpierre — APPROVED** [review]
🎸

**Pull request merged**